integration/package-operator/000_upgrade_test.go

```diff
@@ -36,31 +34,68 @@ const (
 )
 
 func TestUpgrade(t *testing.T) {
-	ctx := logr.NewContext(context.Background(), testr.New(t))
+	log := testr.New(t)
+	ctx := logr.NewContext(context.Background(), log)
+
+	// package that is installed to ensure the teardown job cleans up packages.
+	payloadPkg := &pkocore.Package{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      "success",
+			Namespace: "default",
+		},
+		Spec: pkocore.PackageSpec{
+			Image: SuccessTestPackageImage,
+			Config: &runtime.RawExtension{
+				Raw: []byte(fmt.Sprintf(`{"testStubImage": "%s"}`, TestStubImage)),
+			},
+		},
+	}
 
-	require.NoError(t, deleteExistingPKO(ctx))
+	pkoPackage := &pkocore.ClusterPackage{ObjectMeta: meta.ObjectMeta{Name: PackageOperatorClusterPackageName}}
+	pkoNamespace := &core.Namespace{ObjectMeta: meta.ObjectMeta{Name: "package-operator-system"}}
 
-	log := logr.FromContextOrDiscard(ctx)
-	pkg := &corev1alpha1.ClusterPackage{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "package-operator",
-			Namespace: PackageOperatorNamespace,
-		},
+	selector, err := labels.ValidatedSelectorFromSet(map[string]string{"package-operator.run/package": "package-operator"})
+	if err != nil {
+		panic(err)
+	}
+
+	// Install payload package.
+	require.NoError(t, Client.Create(ctx, payloadPkg))
+	require.NoError(t, Waiter.WaitForCondition(ctx, payloadPkg, pkocore.PackageAvailable, meta.ConditionTrue))
+
+	// Delete existing PKO installation and wait for it to be gone.
+	require.NoError(t, Client.Delete(ctx, pkoPackage))
+
+	// Ensure PKO namespace and package is gone
+	require.NoError(t, Waiter.WaitToBeGone(ctx, pkoNamespace, func(obj client.Object) (done bool, err error) { return }, dev.WithTimeout(30*time.Minute)))
+	require.NoError(t, Waiter.WaitToBeGone(ctx, pkoPackage, func(obj client.Object) (done bool, err error) { return }, dev.WithTimeout(30*time.Minute)))
```

Comment:
I think these timeouts are too high. 5 minutes should be enough.
I also think the timeout should be moved to a const definition at the top of the file for consistency and avoiding magic values.